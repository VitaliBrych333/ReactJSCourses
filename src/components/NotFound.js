import React, { memo } from 'react';
import styled from 'styled-components';

const Styledh3 = styled.h3`
  margin: 150px auto 0;
  text-transform: capitalize;
  text-align: center;
  font-size: 40px;
`;

// PATTERN: Stateless function
const NotFound = () => {
  return <Styledh3>No movie found</Styledh3>;
};

export default memo(NotFound);
