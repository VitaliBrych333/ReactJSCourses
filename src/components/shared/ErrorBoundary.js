import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ErrorBoundary extends Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            textError: null
        };
    }

    componentDidCatch(error, info) {
        console.log('error', error);
        console.log('info', info);
        this.setState({ hasError: true, textError: error.toString() });
    }

    render() {
        if (this.state.hasError) {
            return (
                <div>
                    <p>Error!!!</p>
                    <p>{this.state.textError}</p>
                </div>
            );
        } else {
            return this.props.children;
        }
    }
}

ErrorBoundary.propTypes = {
    children: PropTypes.element
};
