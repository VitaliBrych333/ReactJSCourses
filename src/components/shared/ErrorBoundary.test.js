import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';
import NotFound from '../IncorrectPath';

Enzyme.configure({ adapter: new Adapter() });

describe('<ErrorBoundary/>', () => {
  it('should equals to snapshot of ErrorBoundary', () => {
    const wrapper = mount(
      <ErrorBoundary>
        <NotFound />
      </ErrorBoundary>
    );

    const error = new Error('test');
    wrapper.find(NotFound).simulateError(error);

    expect(wrapper).toMatchSnapshot();
  });
});
