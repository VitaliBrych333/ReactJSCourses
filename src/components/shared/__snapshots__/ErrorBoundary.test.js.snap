// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary/> should equals to snapshot of ErrorBoundary 1`] = `ReactWrapper {}`;
